core/src/main/java/com/datastrato/gravitino/connector/DataTypeConverter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.connector;
+
+import com.datastrato.gravitino.rel.types.Type;
+
+/**
+ * The interface for converting data types between Gravitino and catalogs. In most cases, the ToType
+ * and FromType are the same. But in some cases, such as converting between Gravitino and JDBC
+ * types, the ToType is String and the FromType is JdbcTypeBean.
+ *
+ * @param <ToType> The Gravitino type will be converted to.
+ * @param <FromType> The type will be converted to Gravitino type.
+ */
+public interface DataTypeConverter<ToType, FromType> {
+  /**
+   * Convert the Gravitino type to the catalog type.
+   *
+   * @param type The Gravitino type.
+   * @return The catalog type.
+   */
+  ToType fromGravitino(Type type);
+
+  /**
+   * Convert the catalog type to the Gravitino type.
+   *
+   * @param type The catalog type.
+   * @return The Gravitino type.
+   */
+  Type toGravitino(FromType type);
+}
```

Comment:
I don't see there's much difference here. User still need to a) implement this interface b) create a object to use this interface. The only difference is that it here defines a unified interface.

Ideally, the catalog implementor should only care about how to implement the interface, whereas the other module can leverage this interface. Otherwise, the abstraction is not enough.

Reply:
Yeah, it is difficult to achieve the ideal effect you described for the current catalog interface. 

This PR only defines a unified interface, but there are no constraints on the catalog. However, this is clearer than the previous code. This is the best effort that can be made for a type interface at this time.